### PR TITLE
Fleet

### PR DESCRIPTION
**Pull request by nicolasparada**
Initial fleet support for the CLI.

Added option to create and list the fleets.
Also agents can be updated to be attached to a fleet.
Also agents table show the fleet they belong to.

Missing updating and delete a fleet. Also detaching an agent from a fleet. But those will be for another PR.

~~Waiting https://github.com/calyptia/cloud/issues/809 to be fixed first.~~

### CODE REVIEW

File: cmd/calyptia/generate.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
 package main
 
+//go:generate go run -mod=mod github.com/rjeczalik/interfaces/cmd/interfacer -for github.com/calyptia/api/client.Client -as main.Client -o ./client_gen.go
```
Comment (pwhelan) [inline cmd/calyptia/generate.go:3]: Could we or should we replace moq with interfacer? I don't think we should have two of these kinds of tools.

Comment (nicolasparada) [reply cmd/calyptia/generate.go:3]: interfacer replaces ifacemaker in this case.
ifacemaker uses actual files, while interfacer uses a package, which in this case is the client package.

For context, before we were maintaining the client interface manually. But looks like @niedbalski did generate the client interface using ifacemaker, just I dunno how. Maybe he copied it from the cloud repo manually... dunno.

Comment (pwhelan) [reply cmd/calyptia/generate.go:3]: I think he probably did. That was how it was done before. Someone would generate the files with some magic incantation, commit it to the repo and then... :man_shrugging: 

That's why I purposefully had the generate.go file in calyptlia/cloud install ifacemaker with a specific version. Some versions of ifacemaker work with our repos and some do not. I left ifacemaker out of go.mod/go.sum but I don't remember why... probably to make sure the correct version was always installed by generate.go and not let go modules interfere.

Interfacer seems to be a bit more active. It'd be best to see which project does the best compatibility testing.

Let's leave as is for now and add it to the yak shaving pile?

**Review by pwhelan — APPROVED** [review]
LGTM .. but maybe we should take another look at how and why we generate interfaces later.

File: cmd/calyptia/generate.go (hunk 1 above)
Comment (nicolasparada) [reply cmd/calyptia/generate.go:3]: The thing is that ifacemaker works in cloud but didn't work in this case for the cli.

**Comment by patrick-stephens** [comment]
@nicolasparada can you have a look at the test failures?

**Comment by nicolasparada** [comment]
@patrick-stephens fixed.

**Pull request merged**